### PR TITLE
Feature/add project

### PR DESCRIPTION
**Pull request by marco-gattelli**

### CODE REVIEW

File: kdl-client/src/Components/Stepper/Step.tsx [hunk 1]
```diff
@@ -31,16 +36,23 @@ function Step({ label, completed, error, active, visited, onClick }: Props) {
...
       })}
       onClick={onClick}
     >
       <div className={styles.circle}>
         <Icon className="icon-small" />
       </div>
-      <p className={styles.label}>{label}</p>
+      <p
+        className={cx(styles.label, {
+          [styles.disabled]: !completed && disabled,
```
Comment (mgs95) [inline kdl-client/src/Components/Stepper/Step.tsx:50]: repeated in line 41. You can access this using: `.step.disabled .label`

File: kdl-client/src/Components/Stepper/Step.tsx [hunk 2]
```diff
@@ -31,16 +36,23 @@ function Step({ label, completed, error, active, visited, onClick }: Props) {
     <div
       className={cx(styles.step, {
         [styles.completed]: completed,
-        [styles.error]: error,
```
Comment (mgs95) [inline kdl-client/src/Components/Stepper/Step.tsx:34]: remember to remove this style from the css

File: kdl-client/src/Components/Stepper/Stepper.module.scss [hunk 3]
```diff
@@ -124,4 +130,16 @@ $separator-padding: 3 * $grid-unit;
       border-color: palette(highlight);
     }
   }
+
```
Comment (mgs95) [inline kdl-client/src/Components/Stepper/Stepper.module.scss:133]: at l106, remove error style, not used anymore

File: kdl-client/src/Pages/Cluster/apollo/hooks/useNewProject.ts [hunk 4]
```diff
@@ -22,8 +30,12 @@ function useNewProject(section: keyof NewProject) {
...
 
   function updateError(
-    key: keyof InformationErrors | keyof RepositoryErrors,
-    value: string
+    key:
+      | keyof InformationErrors
+      | keyof RepositoryErrors
+      | keyof ExternalRepositoryErrors
+      | keyof InternalRepositoryErrors,
+    value: any
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/apollo/hooks/useNewProject.ts:38]: errors are always strings (or maybe also boolean), `any` is not neccesary here

File: kdl-client/src/Pages/Cluster/pages/NewProject/NewProject.tsx [hunk 5]
```diff
@@ -65,39 +76,93 @@ function NewProject() {
...
+      data &&
+      data.newProject.externalRepository &&
+      actStep === Steps.REPOSITORY_DETAILS &&
+      type === RepositoryType.EXTERNAL
+    ) {
+      const {
+        values: { isConnectionTested },
+        errors: { warning },
+      } = data.newProject.externalRepository;
+      console.log(
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/NewProject.tsx:132]: remove log

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/RepositoryDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  GetNewProject,
+} from 'Graphql/client/queries/getNewProject.graphql';
+
+import React from 'react';
+import { RepositoryType } from 'Graphql/types/globalTypes';
+import { useQuery } from '@apollo/client';
+import ExternalRepository from './components/ExternalRepository/ExternalRepository';
+import InternalRepository from './components/InternalRepository/InternalRepository';
+
+function RepositoryDetails(params: any) {
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/RepositoryDetails.tsx:13]: any!

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/RepositoryDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    <div>
+      {isExternal ? (
+        <ExternalRepository showErrors={showErrors} />
+      ) : (
+        <InternalRepository showErrors={showErrors} />
+      )}
+    </div>
+  );
+
+  // const repositoryType = type;
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/RepositoryDetails.tsx:35]: remove commented code

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/components/ExternalRepository/ExternalRepository.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+          className={styles.testButton}
+          onClick={validateConnection}
+          disabled={isValidUrl !== true}
+          loading={loading}
+        />
+      </div>
+      {isConnectionTested && !!hasConnectionError && (
+        <>
+          <div className={cx(styles.connectionError, {})}>
+            {'Connection OK.'}
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/components/ExternalRepository/ExternalRepository.tsx:87]: unnecessary brackets, just `Connection OK.`

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/components/InternalRepository/InternalRepository.module.scss [hunk 9]
```diff
@@ -0,0 +1,25 @@
+@import 'Styles/variables';
+@import 'Styles/mixins';
+@import 'Styles/colors';
+
+.repositoryInternal {
+  display: flex;
+  align-items: flex-start;
+  .url {
+    .urlTitle {
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/components/InternalRepository/InternalRepository.module.scss:9]: do not set font properties such as font-size or font-weight, find the font in the design and import it

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/components/InternalRepository/InternalRepository.module.scss [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+      text-transform: uppercase;
+      margin-top: 0;
+      font-size: 11px;
+      color: palette(base, 200);
+      font-weight: 700;
+      line-height: 14px;
+    }
+    .urlContent {
+      margin-top: 2.5 * $grid-unit;
+      font-size: $grid-unit * 2.5;
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/RepositoryDetails/components/InternalRepository/InternalRepository.module.scss:19]: same here

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.module.scss [hunk 11]
```diff
@@ -4,77 +4,68 @@
...
+  .url,
+  .name {
+    margin: 0;
+  }
+  .name {
+    @include font-headline;
+    color: font-color(light);
+  }
+  .url {
+    font-weight: 500;
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.module.scss:27]: use font mixin

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.module.scss [hunk 12]
```diff
@@ -4,77 +4,68 @@
...
+      display: flex;
+      align-items: center;
+    }
+  }
+  .descriptionContainer {
+    margin: 2.5 * $grid-unit 0;
+    background-color: palette(base, 600);
+    color: palette(lowlight, 200);
+    padding: 3 * $grid-unit;
+    font-size: 1.75 * $grid-unit;
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.module.scss:50]: idem

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.tsx [hunk 13]
```diff
@@ -43,39 +39,29 @@ const Section: FC<SectionProps> = ({ title, children }) => (
...
-    },
-  } = data.newProject;
+  const { information, repository } = data.newProject;
+  const type = repository?.values?.type || RepositoryType.EXTERNAL;
+  const isExternalRepo = type === RepositoryType.EXTERNAL;
+  const { name, description } = information.values;
+  const repoTypeDetails = data.newProject[type];
+  const { url } = repoTypeDetails.values;
+
+  console.log(data);
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.tsx:52]: remove log

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.tsx [hunk 14]
```diff
@@ -43,39 +39,29 @@ const Section: FC<SectionProps> = ({ title, children }) => (
...
-        text = 'Connection ok';
-    }
+    const text = hasConnectionError ? 'connection error' : 'connection ok';
 
     return (
       <div
         className={cx(styles.check, {
-          [styles.skip]: skipTest,
-          [styles.error]: connection !== '',
+          [styles.error]: hasConnectionError, //connection !== '',
```
Comment (mgs95) [inline kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.tsx:64]: remove comment

File: kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.tsx (hunk 13 above)
Comment (guumaster) [reply kdl-client/src/Pages/Cluster/pages/NewProject/pages/Summary/Summary.tsx:52]: why don't we use a linter to catch this trivial errors?

**Pull request merged**